### ISSUE TITLE
Rewrite loading state

### ISSUE THREAD

**Issue by khomtali** [redi-js-teachers/js_sprint_2020_final_project_khomtali#39]
@MrTim:
> You can implement this behavior with context. Since you already have this user context you can have loading state there. As well you can have a useReducer hook.
Can recommend you to look at this article: https://kentcdodds.com/blog/how-to-use-react-context-effectively

Please rewrite code of user loading state tracking.

**Comment by BTBTravis** [comment]
I agree contexts would be a good fit. If you are only tracking a single thing with the context I don't think you need use reducer. A simple context with setState would do just fine.
